### PR TITLE
Exclude timestamp from prefix extractor

### PR DESCRIPTION
**Pull request by jay-zhuang**
Timestamp should not be included in prefix extractor, as we discussed here: https://github.com/facebook/rocksdb/pull/7589#discussion_r511068586

Test Plan: added unittest

### CODE REVIEW

File: db/db_with_timestamp_basic_test.cc [hunk 1]
```diff
@@ -909,6 +988,131 @@ TEST_F(DBBasicTestWithTimestamp, CompactDeletionWithTimestampMarkerToBottom) {
...
+  Slice ts = ts_str;
+  write_opts.timestamp = &ts;
+
+  int idx = 0;
+  for (; idx < kMaxKey / 4; idx++) {
+    ASSERT_OK(db_->Put(write_opts, Key1(idx), "bar"));
+    ASSERT_OK(db_->Put(write_opts, KeyWithPrefix("foo", idx), "bar"));
+  }
+
+  Flush();
```
Comment (riversand963) [inline db/db_with_timestamp_basic_test.cc:1033]: `ASSERT_OK`

**Review by riversand963 — APPROVED** [review]
Thanks @jay-zhuang for the PR.

File: db/db_with_timestamp_basic_test.cc [hunk 2]
```diff
@@ -329,6 +353,61 @@ TEST_F(DBBasicTestWithTimestamp, SeekWithPrefixLargerThanKey) {
...
+
+  ASSERT_OK(db_->Put(write_opts, "foo2", "bar"));
+  Flush();
+
+  // Move sst file to next level
+  ASSERT_OK(db_->CompactRange(CompactRangeOptions(), nullptr, nullptr));
+
+  ASSERT_OK(db_->Put(write_opts, "foo3", "bar"));
+  Flush();
```
[the remark this answers is not in the document]
Comment (jay-zhuang) [reply db/db_with_timestamp_basic_test.cc:401]: I'm wondering why STATUS ASSERT test passed?

[the remark this answers is not in the document]
Comment (jay-zhuang) [reply db/db_with_timestamp_basic_test.cc:401]: It's already checked insider of `Flush()`: https://github.com/facebook/rocksdb/blob/1e40696dd14d86b0e0330fd53d8114dc4d9011ad/db/db_impl/db_impl_compaction_flush.cc#L1704-L1706
I guess ideally we should un-check that before return.

**Pull request closed without merging**